test/unit/solver/test_cuda_linear_solver.cpp

```diff
@@ -0,0 +1,164 @@
+        return micm::LinearSolver<double, Group10000SparseVectorMatrix>{ matrix, initial_value };
+      },
+      10000);
+
+  std::vector<double> gpu_x = linearSolverGenerator<Group10000VectorMatrix, Group10000SparseVectorMatrix, micm::CudaLinearSolver<double, Group10000SparseVectorMatrix>>(
+      [](const Group10000SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group10000SparseVectorMatrix> {
+        return micm::CudaLinearSolver<double, Group10000SparseVectorMatrix>{ matrix, initial_value };
+      },
+      10000);
+
+  for (int i = 0; i < cpu_x.size(); i++){
+    EXPECT_EQ(cpu_x, gpu_x); 
+  }
+}
+
+
+TEST(CudaLinearSolver, DenseMatrixVectorOrdering)
+{
+  testDenseMatrix<Group1VectorMatrix, Group1SparseVectorMatrix, micm::CudaLinearSolver<double, Group1SparseVectorMatrix, micm::CudaLuDecomposition>>(
+      [](const Group1SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group1SparseVectorMatrix, micm::CudaLuDecomposition> {
+        return micm::CudaLinearSolver<double, Group1SparseVectorMatrix, micm::CudaLuDecomposition>{matrix, initial_value};
+      });
+}
+
+TEST(CudaLinearSolver, RandomMatrixVectorOrdering)
+{
+  testRandomMatrix<Group1VectorMatrix, Group1SparseVectorMatrix, micm::CudaLinearSolver<double, Group1SparseVectorMatrix>>(
+      [](const Group1SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group1SparseVectorMatrix> {
+        return micm::CudaLinearSolver<double, Group1SparseVectorMatrix>{ matrix, initial_value };
+      },
+      1);
+  testRandomMatrix<Group2VectorMatrix, Group2SparseVectorMatrix, micm::CudaLinearSolver<double, Group2SparseVectorMatrix>>(
+      [](const Group2SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group2SparseVectorMatrix> {
+        return micm::CudaLinearSolver<double, Group2SparseVectorMatrix>{ matrix, initial_value };
+      },
+      2);
+  testRandomMatrix<Group3VectorMatrix, Group3SparseVectorMatrix, micm::CudaLinearSolver<double, Group3SparseVectorMatrix>>(
+      [](const Group3SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group3SparseVectorMatrix> {
+        return micm::CudaLinearSolver<double, Group3SparseVectorMatrix>{ matrix, initial_value };
+      },
+      3);
+  testRandomMatrix<Group4VectorMatrix, Group4SparseVectorMatrix, micm::CudaLinearSolver<double, Group4SparseVectorMatrix>>(
+      [](const Group4SparseVectorMatrix<double>& matrix,
+         double initial_value) -> micm::CudaLinearSolver<double, Group4SparseVectorMatrix> {
+        return micm::CudaLinearSolver<double, Group4SparseVectorMatrix>{ matrix, initial_value };
+      },
+      4);
+  gpuValidation();
+  
+}
+
+TEST(CudaLinearSolver, DiagonalMatrixVectorOrdering)
```

Comment:
I do not think that the matrix for chemical reactions is diagonal in general. What is this test for?